Feature | Extend Swagger Coverage for controller `OAuth2PaymentGatewayProfileApiController`

matiasperrone-exo opened the pull request:
## Task:
Ref: https://app.clickup.com/t/86b6wkgxj

caseylocker reviewed (changes requested):
@matiasperrone-exo Looking at this I believe that the 5 occurences of "oauth2_security_scope" should be "summit_rsvp_oauth2".

Change all 5 occurrences from:
security: [['oauth2_security_scope' => [SummitScopes::WriteSummitData]]],
To:
security: [['summit_rsvp_oauth2' => [SummitScopes::WriteSummitData]]],

matiasperrone-exo commented:
> @matiasperrone-exo Looking at this I don't see "oauth2_security_scope" defined anywhere.

@caseylocker thanks! I added a new security schema and replaced with the correct scopes

smarcet commented on app/Swagger/schemas.php:
```diff
@@ -351,3 +351,93 @@ class RSVPUpdateRequestSchema_{
+)]
+class PaymentGatewayProfileSchema {}
+
+#[OA\Schema(
```
@matiasperrone-exo why are u defining these here ?
u need to create a file schema per controller and define there this as we already spoke

caseylocker reviewed (changes requested):
@matiasperrone-exo The format of the security definitions is incorrect. Missing inner array brackets around the scopes:
// Current (INVALID)
security: [['summit_payment_gateway_oauth2' =>
    SummitScopes::ReadAllSummitData,
    SummitScopes::ReadPaymentProfiles
]],

// Required (CORRECT)
security: [['summit_payment_gateway_oauth2' => [
    SummitScopes::ReadAllSummitData,
    SummitScopes::ReadPaymentProfiles
]]],

We're also missing the operationId on all endpoints. Endpoints and suggestions...


GET /api/v1/summits/{id}/payment-gateway-profiles | getAllPaymentGatewayProfiles
-- | --
GET /api/v1/summits/{id}/payment-gateway-profiles/{payment_profile_id} | getPaymentGatewayProfile
POST /api/v1/summits/{id}/payment-gateway-profiles | createPaymentGatewayProfile
PUT /api/v1/summits/{id}/payment-gateway-profiles/{payment_profile_id} | updatePaymentGatewayProfile
DELETE /api/v1/summits/{id}/payment-gateway-profiles/{payment_profile_id} | deletePaymentGatewayProfile

matiasperrone-exo commented:
Thanks @caseylocker for the comments. Now is ready to review again.

I fixed the missing brackets and the rebase conflicts
